tests/org.palladiosimulator.dataflow.confidentiality.analysis.tests/src/org/palladiosimulator/dataflow/confidentiality/analysis/constraint/ConstraintResultTest.java

```diff
@@ -0,0 +1,192 @@
+     */
+    private Stream<Arguments> provideTestConstraintResults() {
+    	Predicate<AbstractActionSequenceElement<?>> travelPlannerConstraint = node -> travelPlannerCondition(node);
+    	Predicate<AbstractActionSequenceElement<?>> internationalOnlineShopConstraint = node -> internationalOnlineShopCondition(node);
+
+    	StandalonePCMDataFlowConfidentialtyAnalysis mutipleAssembliesAnalysis = 
+    			super.initializeAnalysis(Paths.get("models", "MultipleAssembliesTest", "default.usagemodel"), Paths.get("models", "MultipleAssembliesTest", "default.allocation"));
+    	
+    	StandalonePCMDataFlowConfidentialtyAnalysis dataStoreAnalysis = 
+    			super.initializeAnalysis(Paths.get("models", "DatastoreTest", "default.usagemodel"), Paths.get("models", "DatastoreTest", "default.allocation"));
+    	Predicate<AbstractActionSequenceElement<?>> dataStoreConstraint = node -> dataStoreCondition(node);
+    	
+    	StandalonePCMDataFlowConfidentialtyAnalysis returnAnalysis = 
+    			super.initializeAnalysis(Paths.get("models", "ReturnTestModel", "default.usagemodel"), Paths.get("models", "ReturnTestModel", "default.allocation"));
+    	Predicate<AbstractActionSequenceElement<?>> returnConstraint = node -> returnCondition(node);
+    	
+    	return Stream.of(
+    			Arguments.of(travelPlannerAnalysis, travelPlannerConstraint, ConstraintViolations.travelPlannerViolations),
+    			Arguments.of(internationalOnlineShopAnalysis, internationalOnlineShopConstraint, ConstraintViolations.internationalOnlineShopViolations),
+    			Arguments.of(mutipleAssembliesAnalysis, internationalOnlineShopConstraint, ConstraintViolations.multipleAssembliesViolations),
+    			Arguments.of(dataStoreAnalysis, dataStoreConstraint, ConstraintViolations.dataStoreViolations),
+    			Arguments.of(returnAnalysis, returnConstraint, ConstraintViolations.returnViolations)
+ 
+    	);
+    }
+    
+    /**
+     * Tests, whether the analysis correctly identifies violations for the example models
+     * <p>
+     * Fails if the analysis does not propagate the correct characteristics for each ActionSequence
+     */
+    @ParameterizedTest
+    @MethodSource("provideTestConstraintResults")
+    @DisplayName("Test results of the analysis with a constraint")
+    public void testContraintResults(StandalonePCMDataFlowConfidentialtyAnalysis analysis, Predicate<AbstractActionSequenceElement<?>> constraint, List<ConstraintData> constraintData) {
+    	analysis.setLoggerLevel(Level.TRACE);
+    	List<ActionSequence> actionSequences = analysis.findAllSequences();
+    	List<ActionSequence> evaluatedSequences = analysis.evaluateDataFlows(actionSequences);
+    	
+    	List<AbstractActionSequenceElement<?>> results = evaluatedSequences.stream()
+    			.map(it -> analysis.queryDataFlow(it, constraint))
+    			.flatMap(it -> it.stream())
+    			.collect(Collectors.toList());
+    	
+    	assertEquals(constraintData.size(), results.size(), "Incorrect amount of violations found");
```

Comment:
Here, but also at other parts of the code (e.g. in the data classes): The word `amount` usually specifies an uncountable number of items, while `count` refers to countable sets. See, e.g., this thread [here](https://ell.stackexchange.com/questions/3511/whats-the-difference-between-words-number-count-amount-and-quantity)